tools/nnyeah/nnyeah/Program.cs

```diff
@@ -1,23 +1,92 @@
 using System;
 using System.IO;
+using System.Reflection;
+using Mono.Options;
+using System.Collections.Generic;
 
 #nullable enable
 
 namespace nnyeah {
-    class Program {
-        static void Main (string [] args)
-        {
-            if (args.Length != 2) {
-                Console.Error.WriteLine ("Usage: dotnet nnyeah.dll /path/to/input/file.dll /path/to/output/file.dll");
-                Environment.Exit (1);
-            }
-            using var stm = new FileStream (args [0], FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
-            var reworker = new Reworker (stm);
-            reworker.Load ();
-            if (reworker.NeedsReworking ()) {
-                using var ostm = new FileStream (args [1], FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite);
-                reworker.Rework (ostm);
-            }
-        }
-    }
-}    
+	class Program {
+		static void Main (string [] args)
+		{
+			var doHelp = false;
+			string? infile = null, outfile = null;
+			var verbose = false;
+			var forceOverwrite = false;
+			var suppressWarnings = false;
+			var warnings = new List<string> ();
+			var transforms = new List<string> ();
+
+			var options = new OptionSet () {
+				{ "h|?|help", o => doHelp = true },
+				{ "i=|input=", f => infile = f },
+				{ "o=|output=", f => outfile = f },
+				{ "v|verbose", o => verbose = true },
+				{ "f|force-overwrite", o => forceOverwrite = true },
+				{ "s|suppress-warnings", o => suppressWarnings = true },
+			};
+
+			try {
+				var extra = options.Parse (args);
+			} catch {
+				doHelp = true;
+			}
+
+
+			if (doHelp || infile is null || outfile is null) {
+				PrintOptions (options, Console.Out);
+				Environment.Exit (0);
+			}
+
+			if (!File.Exists (infile)) {
```

Comment:
This is not taking into account a path that is a directory, I have created an issue to fix it in the next PR: https://github.com/xamarin/xamarin-macios/issues/14584